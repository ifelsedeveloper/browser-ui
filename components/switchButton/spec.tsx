/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import SwitchButton from '../switchButton'

describe.skip('switchButton tests', () => {
  const baseComponent = (props?: object) => (
    <SwitchButton id='switchButton' checked={false} {...props} />
  )

  it('matches the snapshot', () => {
    const component = baseComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(baseComponent({id: 'switchButton'}))
    const assertion = wrapper.find('#switchButton').length
    expect(assertion).toBe(1)
  })

  it('can be checked', () => {
    const wrapper = shallow(baseComponent({checked: true}))
    const assertion = wrapper.find('input').props().checked
    expect(assertion).toEqual(true)
  })

  it('can be unchecked', () => {
    const wrapper = shallow(baseComponent({checked: false}))
    const assertion = wrapper.find('input').props().checked
    expect(assertion).toEqual(false)
  })

  it('can be disabled', () => {
    const wrapper = shallow(baseComponent({disabled: true}))
    const assertion = wrapper.find('input').props().disabled
    expect(assertion).toBe(true)
  })

  it('can be set as read-only', () => {
    const wrapper = shallow(baseComponent({readOnly: true}))
    const assertion = wrapper.find('input').props().readOnly
    expect(assertion).toBe(true)
  })

  it('can have autoFocus', () => {
    const wrapper = shallow(baseComponent({autoFocus: true}))
    const assertion = wrapper.find('input').props().autoFocus
    expect(assertion).toBe(true)
  })

  it('can have left text as label', () => {
    const wrapper = shallow(baseComponent({leftText: 'LEFT TEXT FTW'}))
    const assertion = wrapper.find('#switchButtonLeftText').text()
    expect(assertion).toBe('LEFT TEXT FTW')
  })

  it('can have right text as label', () => {
    const wrapper = shallow(baseComponent({rightText: 'RIGHT TEXT FTW'}))
    const assertion = wrapper.find('#switchButtonRightText').text()
    expect(assertion).toBe('RIGHT TEXT FTW')
  })

  it('can respond to onChange', () => {
    const value = {target: { checked: true }}
    const onChange = jest.fn()
    const wrapper = shallow(baseComponent({onChange}))
    wrapper.find('input').simulate('change', value)
    expect(onChange).toBeCalledWith(value)
  })
})
