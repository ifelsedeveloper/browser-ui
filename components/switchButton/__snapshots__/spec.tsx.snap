// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switchButton tests matches the snapshot 1`] = `
<div
  className="switchButton_wrapper"
>
  <div>
    <input
      autoFocus={undefined}
      checked={false}
      className="checkbox"
      disabled={false}
      id="switchButton"
      onChange={[Function]}
      readOnly={undefined}
      type="checkbox"
    />
    <label
      className="switchButton"
      htmlFor="switchButton"
      style={
        Object {
          "--height": false,
          "--width": false,
        }
      }
    >
      <i
        className="switchButton_knob"
        style={
          Object {
            "--height": false,
            "--width": false,
          }
        }
      />
    </label>
  </div>
</div>
`;
